from src import *
from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_FileEncWindow(object):
    def setupUi(self, FileEncWindow):
        FileEncWindow.setObjectName("FileEncWindow")
        FileEncWindow.resize(500, 300)
        FileEncWindow.setAcceptDrops(False)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("program\src\icon\icon.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        FileEncWindow.setWindowIcon(icon)
        self.centralwidget = QtWidgets.QWidget(FileEncWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.title_label = QtWidgets.QLabel(self.centralwidget)
        self.title_label.setGeometry(QtCore.QRect(10, 10, 481, 51))
        font = QtGui.QFont()
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        font.setKerning(True)
        self.title_label.setFont(font)
        self.title_label.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.title_label.setMouseTracking(False)
        self.title_label.setTextFormat(QtCore.Qt.AutoText)
        self.title_label.setAlignment(QtCore.Qt.AlignCenter)
        self.title_label.setObjectName("title_label")
        self.file_enc_btn = QtWidgets.QPushButton(self.centralwidget)
        self.file_enc_btn.setGeometry(QtCore.QRect(110, 210, 280, 50))
        self.file_enc_btn.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.file_enc_btn.setObjectName("file_enc_btn")
        self.file_path_label = QtWidgets.QLabel(self.centralwidget)
        self.file_path_label.setGeometry(QtCore.QRect(110, 70, 240, 50))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.file_path_label.sizePolicy().hasHeightForWidth())
        self.file_path_label.setSizePolicy(sizePolicy)
        self.file_path_label.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.file_path_label.setAcceptDrops(False)
        self.file_path_label.setAlignment(QtCore.Qt.AlignCenter)
        self.file_path_label.setObjectName("file_path_label")
        self.file_select_btn = QtWidgets.QToolButton(self.centralwidget)
        self.file_select_btn.setGeometry(QtCore.QRect(360, 70, 30, 50))
        self.file_select_btn.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.file_select_btn.setPopupMode(QtWidgets.QToolButton.DelayedPopup)
        self.file_select_btn.setObjectName("file_select_btn")
        self.pw_entry = QtWidgets.QLineEdit(self.centralwidget)
        self.pw_entry.setGeometry(QtCore.QRect(110, 131, 281, 20))
        self.pw_entry.setText("")
        self.pw_entry.setMaxLength(32767)
        self.pw_entry.setFrame(True)
        self.pw_entry.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.pw_entry.setDragEnabled(True)
        self.pw_entry.setCursorMoveStyle(QtCore.Qt.LogicalMoveStyle)
        self.pw_entry.setClearButtonEnabled(True)
        self.pw_entry.setObjectName("pw_entry")
        self.pw_cf_entry = QtWidgets.QLineEdit(self.centralwidget)
        self.pw_cf_entry.setGeometry(QtCore.QRect(110, 160, 281, 20))
        self.pw_cf_entry.setText("")
        self.pw_cf_entry.setMaxLength(32767)
        self.pw_cf_entry.setFrame(True)
        self.pw_cf_entry.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.pw_cf_entry.setDragEnabled(True)
        self.pw_cf_entry.setCursorMoveStyle(QtCore.Qt.LogicalMoveStyle)
        self.pw_cf_entry.setClearButtonEnabled(True)
        self.pw_cf_entry.setObjectName("pw_cf_entry")
        self.pw_chk_box = QtWidgets.QCheckBox(self.centralwidget)
        self.pw_chk_box.setGeometry(QtCore.QRect(110, 190, 280, 14))
        self.pw_chk_box.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.pw_chk_box.setObjectName("pw_chk_box")
        self.enc_progress_bar = QtWidgets.QProgressBar(self.centralwidget)
        self.enc_progress_bar.setEnabled(True)
        self.enc_progress_bar.setGeometry(QtCore.QRect(110, 210, 280, 50))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.enc_progress_bar.sizePolicy().hasHeightForWidth())
        self.enc_progress_bar.setSizePolicy(sizePolicy)
        self.enc_progress_bar.setAutoFillBackground(False)
        self.enc_progress_bar.setProperty("value", 24)
        self.enc_progress_bar.setTextVisible(False)
        self.enc_progress_bar.setOrientation(QtCore.Qt.Horizontal)
        self.enc_progress_bar.setInvertedAppearance(False)
        self.enc_progress_bar.setTextDirection(QtWidgets.QProgressBar.TopToBottom)
        self.enc_progress_bar.setObjectName("enc_progress_bar")
        self.enc_progress_bar.raise_()
        self.title_label.raise_()
        self.file_enc_btn.raise_()
        self.file_path_label.raise_()
        self.file_select_btn.raise_()
        self.pw_entry.raise_()
        self.pw_cf_entry.raise_()
        self.pw_chk_box.raise_()
        FileEncWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(FileEncWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 500, 21))
        self.menubar.setObjectName("menubar")
        FileEncWindow.setMenuBar(self.menubar)

        self.retranslateUi(FileEncWindow)
        QtCore.QMetaObject.connectSlotsByName(FileEncWindow)

    def retranslateUi(self, FileEncWindow):
        _translate = QtCore.QCoreApplication.translate
        FileEncWindow.setWindowTitle(_translate("FileEncWindow", "program_title"))
        self.title_label.setText(_translate("FileEncWindow", "program_title"))
        self.file_enc_btn.setText(_translate("FileEncWindow", "File Encryption"))
        self.file_path_label.setText(_translate("FileEncWindow", "Drag and drop file here"))
        self.file_select_btn.setText(_translate("FileEncWindow", "..."))
        self.pw_entry.setPlaceholderText(_translate("FileEncWindow", "Enter file password"))
        self.pw_cf_entry.setPlaceholderText(_translate("FileEncWindow", "Enter file password (confirm)"))
        self.pw_chk_box.setText(_translate("FileEncWindow", "I remembered my password.                            "))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    FileEncWindow = QtWidgets.QMainWindow()
    ui = Ui_FileEncWindow()
    ui.setupUi(FileEncWindow)
    FileEncWindow.show()
    sys.exit(app.exec_())
